emote/memory/memory.py

```diff
@@ -463,6 +463,38 @@ def __iter__(self):
             yield {self.data_group: data, self.size_key: data[self.size_key]}
 
 
+class JointMemoryLoader:
+    """A memory loader capable of loading data from multiple `MemoryLoader`s.
+
+    If a datagroup is specified via the `data_group` param, it will place all loaded data in the specified datagroup,
+    otherwise it will directly return the data fetched from each individual loader.
+    """
+
+    def __init__(self, loaders: list[MemoryLoader], data_group: str | None = None):
+        self._loaders = loaders
+        self._data_group = data_group
+
+    def is_ready(self):
+        return all([loader.is_ready() for loader in self._loaders])
```

Comment:
Nit: Should be fine to pass a generator instead of using a listcomp.

Reply:
Very nice, thanks. I did not think of it.